gossipd/gossip_store.c

```diff
@@ -158,13 +158,56 @@ static bool gossip_store_append(int fd,
 		write(fd, msg, msglen) == msglen);
 }
 
+/* Copy a whole message from one gossip_store to another.  Returns
+ * total msg length including header, or 0 on error. */
+static size_t copy_message(int in_fd, int out_fd, unsigned offset)
+{
+	beint32_t belen, becsum;
+	u32 msglen;
+	u8 *msg;
+
+	/* FIXME: optimize both read and allocation */
```

Comment:
why not use `pread` here as well?

Reply:
Because I didn't have time to do all the optimizations.... I didn't even benchmark the final ones :(